acceptance-tests/src/main/kotlin/org/ostelco/at/jersey/Tests.kt

```diff
@@ -1897,7 +1899,157 @@ class PlanTest {
             purchaseRecords.sortBy { it.timestamp }
 
             assert(Instant.now().toEpochMilli() - purchaseRecords.last().timestamp < 10_000) { "Missing Purchase Record" }
-            assertEquals(expectedPlanProduct, purchaseRecords.last().product, "Incorrect 'Product' in purchase record")
+            assertEquals(expectedPlanProductSG, purchaseRecords.last().product, "Incorrect 'Product' in purchase record")
+        } finally {
+            StripePayment.deleteCustomer(customerId = customerId)
+        }
+    }
+}
+
+class RenewPlanTest {
+
+    @Test
+    fun `jersey test - POST purchase plan with trial time`() {
+
+        val email = "purchase-${randomInt()}@test.com"
+        val sku = "PLAN_10USD_DAY"
+        var customerId = ""
+
+        try {
+            customerId = createCustomer(name = "Test Purchase Plan User", email = email).id
+            enableRegion(email = email, region = "us")
+
+            val sourceId = StripePayment.createPaymentTokenId()
+
+            post<String> {
+                path = "/products/$sku/purchase"
+                this.email = email
+                queryParams = mapOf("sourceId" to sourceId)
+            }
+
+            Thread.sleep(200) // wait for 200 ms for balance to be updated in db
+
+            var purchaseRecords: PurchaseRecordList = get {
+                path = "/purchases"
+                this.email = email
+            }
+            purchaseRecords.sortBy { it.timestamp }
+
+            assert(Instant.now().toEpochMilli() - purchaseRecords.last().timestamp < 10_000) { "Missing purchase record" }
+
+            // First record - free product
+            // Second record - product subscribed to
+            assert(2 == purchaseRecords.size) { "Got ${purchaseRecords.size} purchase records, expected 2" }
+            assertEquals(expectedPlanProductUS, purchaseRecords.last().product, "Incorrect 'Product' in purchase record")
+
+            // Actual charge for renewal will first be done after trial time
+            // expires, which will happen after 4 sec. Waiting a bit longer
+            // before checking the outcome.
+            StripeEventListener.waitForSubscriptionPaymentToSucceed(
+                    subscription = "stripe-event-jersey-purchase-ok-sub",
```

Comment:
What does this magic string mean? Consider putting it into a variable with a descriptive name.

Reply:
Name of a  subscription to the "stripe-event" pub-sub topic. We need one such subscription for each "recurring payment" acceptance test. The  `waitForSubscriptionPaymentToSucceed` method is there to ensure that we don't wait too long for the expected events.

Not sure if it make sense use a variable for the subscription name. But some more comments about what is going on could be an idea.